group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilder.java

```diff
@@ -277,13 +277,22 @@ public TargetAssignmentBuilder removeMember(
      */
     public TargetAssignmentResult build() throws PartitionAssignorException {
         Map<String, AssignmentMemberSpec> memberSpecs = new HashMap<>();
+        Map<String, Map<Uuid, Set<Integer>>> assignedPartitions = new HashMap<>(members.size());
```

Comment:
Hum... Do we really need this? Would it be possible to pass `targetAssignment` to GroupSpecImpl like we did with `invertedTargetAssignment`? This will save us from having to create a new HashMap with the same data.

Reply:
What about the static member case?